### PR TITLE
feat: add configure Github deployment action

### PR DESCRIPTION
**Pull request by evan-boone**
Adds action to configure Github deployment

### CODE REVIEW

**Review by stekern — APPROVED** [review]
LGTM 🎉 

It's worth mentioning that the new tests kind of break the testing pattern wrt. isolation as the test for updating a deployment relies on the result of a previous test. It's a bit tricky to avoid here, but it's something I would try to avoid where we can to avoid unnecessarily coupling the tests (e.g., by striving to limit one "arrange-act-assert" instance to a single test case).

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -462,3 +463,47 @@ jobs:
...
           ASSERT_GOT: ${{ steps.act-trigger-deployment-pipeline-cdk-source-should-fail.outcome }}
         run: test "$ASSERT_GOT" = "$ASSERT_EXPECTED"
+      ######################################################
+      #
+      # test: configure-github-deployment
+      #
+      ######################################################
+      - uses: ./configure-github-deployment
+        name: act-configure-github-deployment-should-fail
+        id: act-configure-github-deployment-should-fail
```
[the remark this answers is not in the document]
Comment (evan-boone) [reply .github/workflows/tests.yml:473]: Good point. Removing it!

**Review by joakimen — CHANGES_REQUESTED** [review]
The code looks correct 👍🏻 Just need to consider the workflow permissions for the test workflow.

I do think we are conflating the use of `state` and `status` a bit; according to the [documentation](https://docs.github.com/en/rest/deployments/statuses?apiVersion=2022-11-28#create-a-deployment-status), a deployment may have one or more statuses, and a status has a state (`.deployment.status[].state`). I think it'd be more natural to refer to it as the "deployment status" instead of the "deployment state", as this feels more aligned with the API (in which state is just one of the properties of a status, and not a property of a deployment)

It's subjective though, and not worth the rewrite. Just crossed my mind during review

**Comment by joakimen** [comment]
Another thing that came up during review is that actionlint complains about several of our actions (parse-config, ..) not having documented output parameters. Will create a task to address that in a separate PR

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -13,10 +13,6 @@ concurrency: ci-${{ github.ref }}-${{ github.workflow }}
 jobs:
   test:
     uses: ./.github/workflows/tests.yml
-    permissions:
-      actions: read
-      id-token: write
-      contents: read
```
[the remark this answers is not in the document]
Comment (evan-boone) [reply .github/workflows/ci.yml:19]: Yes, it seems to be necessary! Adding them again

**Comment by evan-boone** [comment]
> The code looks correct 👍🏻 Just need to consider the workflow permissions for the test workflow.
> 
> I do think we are conflating the use of `state` and `status` a bit; according to the [documentation](https://docs.github.com/en/rest/deployments/statuses?apiVersion=2022-11-28#create-a-deployment-status), a deployment may have one or more statuses, and a status has a state (`.deployment.status[].state`). I think it'd be more natural to refer to it as the "deployment status" instead of the "deployment state", as this feels more aligned with the API (in which state is just one of the properties of a status, and not a property of a deployment)
> 
> It's subjective though, and not worth the rewrite. Just crossed my mind during review

Good point, but since the only field we actually allow the consumer to configure is the newest state (which is updated through updating a status), I think it is a bit clearer the way it is written now. If we referred to deployment status, it might seem like the action allows all fields belonging to a status object (including f.eks. `description`) to be configured.

**Review by joakimen — APPROVED** [review]

**Pull request merged**